core/src/main/scala/kafka/server/ReplicaFetcherThread.scala

```diff
@@ -72,7 +72,7 @@ class ReplicaFetcherThread(name: String,
       metrics,
       time,
       "replica-fetcher",
-      Map("broker-id" -> sourceBroker.id.toString).asJava,
+      Map("thread-id" -> name).asJava,
```

Comment:
The fact that the broker id is part of the thread name is a bit of an implementation detail also.